ethereum/contracts/ScaleCodec.sol

```diff
@@ -11,6 +11,11 @@ library ScaleCodec {
         return number;
     }
 
+    function encodeUintCompact(uint16 number) public pure returns (bytes2) {
+        // TODO - remove hardcoded implementation
+        return bytes2(uint16(0xc101));
```

Comment:
If we leave the function in here, we should indeed use a working generic implementation. 

For the MMR leaf, this is not necessary though, since the length of that type is fixed. The easiest workaround is to just use the fixed length prefix as a constant in the `encodeMMRLeaf` function in `ethereum/contracts/BeefyLightClient.sol`.

Reply:
https://github.com/paritytech/polkadot/blob/7ab51d455d11bdaeacd923b279b912b92f0bc580/runtime/common/src/mmr.rs#L65 yeh looks like it is

Reply:
https://github.com/Snowfork/polkadot-ethereum/pull/442/commits/2f82fc9240be1128bbbce0807e5b1f76fdd303be